docs/src/audio.js

```diff
@@ -33,47 +38,61 @@
   Audio.prototype.initializeMicrophoneSampling = function () {
     var errorCallback = function (err) {
       // We should fallback to an audio file here, but that's difficult on mobile
-      if (_this.context.state === 'suspended') {
+      if (_this.context.state === "suspended") {
         var resume = function () {
           _this.context.resume();
 
           setTimeout(function () {
-            if (_this.context.state === 'running') {
-              document.body.removeEventListener('touchend', resume, false);
+            if (_this.context.state === "running") {
+              document.body.removeEventListener("touchend", resume, false);
             }
           }, 0);
         };
 
-        document.body.addEventListener('touchend', resume, false);
-      };
+        document.body.addEventListener("touchend", resume, false);
+      }
     };
 
     try {
-      navigator.getUserMedia = navigator.webkitGetUserMedia ||
-        navigator.getUserMedia || navigator.mediaDevices.getUserMedia;
-      var constraints = { video: false, audio: true };
-      var successCallback = function (mediaStream) {
-        document.getElementById('audioControl').style.display = 'none';
-        console.log('User allowed microphone access.');
-        console.log('Initializing AudioNode from MediaStream');
+      navigator.getUserMedia =
+        navigator.webkitGetUserMedia ||
+        navigator.getUserMedia ||
+        navigator.mediaDevices.getUserMedia;
+      var constraints = {
+        video: false,
+        audio: true,
+      };
+      var successCallback = function successCallback(mediaStream) {
+        document.getElementById("audioControl").style.display = "none";
+        console.log("User allowed microphone access.");
+        console.log("Initializing AudioNode from MediaStream");
         var source = _this.context.createMediaStreamSource(mediaStream);
-        console.log('Setting Meyda Source to Microphone');
+        console.log("Setting Meyda Source to Microphone");
         _this.meyda.setSource(source);
         console.groupEnd();
       };
 
-      console.log('Asking for permission...');
-      let getUserMediaPromise = navigator.getUserMedia(
-        constraints,
-        successCallback,
-        errorCallback
-      );
-      if (getUserMediaPromise) {
-        p.then(successCallback);
-        p.catch(errorCallback);
-      }
-    }
-    catch (e) {
+      console.log("Asking for permission...");
+      var getUserMediaPromise = function (error) {
```

Comment:
This shouldn't be called a promise if it's not a promise anymore. This is a Promise.catch callback.

I think the core logic change here is that if the call to `navigator.mediaDevices.getUserMedia` fails, then in the Promise.catch callback it tries again with `navigator.getUserMedia`. But that should already be handled on line 60 above, so I don't really get this change. I think I'll need to do another commit to split out the prettier changes from the logic changes so that it's clearer.